appservice/src/deleteSite/DeleteSiteStep.ts

```diff
@@ -0,0 +1,41 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.txt in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+
+import { AzureWizardExecuteStep, nonNullProp } from "@microsoft/vscode-azext-utils";
+import { Progress } from "vscode";
+import { ext } from "../extensionVariables";
+import { localize } from "../localize";
+import { IDeleteSiteWizardContext } from "./IDeleteSiteWizardContext";
+
+export class DeleteSiteStep extends AzureWizardExecuteStep<IDeleteSiteWizardContext> {
+    public priority: number = 100;
+
+    public async execute(context: IDeleteSiteWizardContext, progress: Progress<{ message?: string | undefined; increment?: number | undefined; }>): Promise<void> {
+
+        const site = nonNullProp(context, "site");
+
+        const deleting: string = site.isSlot ?
```

Comment:
This is a little tough to follow, might be better to do something like this:

```typescript
let deleting: string;
let deleteSucceeded: string;

if (site.isSlot) {
    deleting = ...
    deleteSucceeded = ...
} else if (site.isFunctionApp) {
    ...
} else {
    ...
}
```

Reply:
Agreed, and fixed 🤠